src/crimson/os/seastore/btree/fixed_kv_node.h

```diff
@@ -218,7 +218,12 @@ struct FixedKVInternalNode
     LOG_PREFIX(FixedKVInternalNode::resolve_relative_addrs);
     for (auto i: *this) {
       if (i->get_val().is_relative()) {
-	auto updated = base.add_relative(i->get_val());
+	paddr_t updated;
+	if (base.get_addr_type() == addr_types_t::SEGMENT) {
+	  updated = base.add_relative(i->get_val());
+	} else {
+	  updated = base.add_offset(i->get_val().as_seg_paddr().get_segment_off());
```

Comment:
add_relative has well-defined behavior for both segmented and rbm addresses.  Fix paddr_t::add_relative to behave correctly regardless of the type so that we don't need to add these branches whenever we need to use add_relative.